ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -1018,3 +1035,20 @@ func compareRayClusterJsonHash(spec1 rayv1alpha1.RayClusterSpec, spec2 rayv1alph
 	}
 	return hash1 == hash2, nil
 }
+
+// getHeadPod returns the head pod of the RayCluster.
+func (r *RayServiceReconciler) isHeadPodRunningAndReady(instance *rayv1alpha1.RayCluster) (bool, error) {
+	podList := corev1.PodList{}
+	filterLabels := client.MatchingLabels{common.RayClusterLabelKey: instance.Name, common.RayNodeTypeLabelKey: string(rayv1alpha1.HeadNode)}
+
+	if err := r.List(context.TODO(), &podList, client.InNamespace(instance.Namespace), filterLabels); err != nil {
+		r.Log.Error(err, "Failed to list the head Pod of the RayCluster %s in the namespace %s", instance.Name, instance.Namespace)
+		return false, err
+	}
+
+	if len(podList.Items) != 1 {
```

Comment:
when do we expect this case? having multiple head nodes?

Reply:
Yes, in KubeRay, it is impossible to have more than one head Pod for a RayCluster. If there is more than one Pod that fulfills the matching labels, it may be caused by either a KubeRay bug or a user misconfiguration.

```
client.MatchingLabels{common.RayClusterLabelKey: instance.Name, common.RayNodeTypeLabelKey: string(rayv1alpha1.HeadNode)}
```